### ISSUE TITLE
CONVERT BAM TO U-BAM Error

### ISSUE THREAD

**Issue by iumsuser** [broadinstitute/picard#1872]
CONVERT BAM TO U-BAM using Picard RevertSam Error

----

### Description 

I'm struggling on how to fix the “Picard's command line syntax is changing.” Error on Linux.

The old code line:

for f in ${OUTPUT_DIR}/*aligned.bam; do picard RevertSam I=$f O=${f/aligned.bam/unaligned.bam} ATTRIBUTE_TO_CLEAR=XS ATTRIBUTE_TO_CLEAR=XA ; done

I've visited the suggested link address (https://lnkd.in/e5sqTNmA), https://lnkd.in/eGc7B_4s) and tried variety of possible code lines, but nothing changed.
I sincerely appreciate your guidance
![Screenshot (2833)](https://user-images.githubusercontent.com/132023067/235011918-91fa7c25-67b5-4116-af53-b80d9e535c39.png)

----

### Tool(s) involved
picard , revertsam

----

**Comment by cmnbroad** [comment]
@iumsuser The “Picard's command line syntax is changing.”  is no something you need to fix - it is always displayed; is a warning only; and it can be safely ignored. As for why your RevertSam command itself is failing, that appears to be due to the way your shell variable expansion/substitution is written, but 'm not enough of a shell expert to tell you how to fix that. I suspect if you supplied the file name directly to the command (rather than though a shell variable), it would work. If not, please post an example, including your literal command, and we can take another look.

**Comment by iumsuser** [comment]
Yes exactly, The solution was introducing input file correctly! thanks a lot!!
